### PR TITLE
feat: Integrating Server sent events for Log streaming and Execution progress

### PR DESCRIPTION
**Pull request by vardanbansal-harness**
Integrating Server sent events for Log streaming and Execution progress

### CODE REVIEW

File: apps/gitness/src/framework/hooks/useLogs.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+interface UseLogsInterface {
+  repoPath: string
+  pipelineId: string
+  stageNum: number
+  stepNum: number
+  stepStatus?: ExecutionState
+  executionNum: string
+}
+
+export const useLogs = ({
```
Comment (cjlee01) [inline apps/gitness/src/framework/hooks/useLogs.ts:22]: should i similarily implement this into the usespacesse ? @vardanbansal-harness

File: apps/gitness/src/framework/hooks/useLogs.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+interface UseLogsReturnType {
+  logs: LivelogLine[]
+}
+
+enum LogsSSEEvent {
+  MESSAGE = 'message'
+}
+
+interface UseLogsInterface {
```
Comment (cjlee01) [inline apps/gitness/src/framework/hooks/useLogs.ts:13]: add this to a interface file?

File: apps/gitness/src/framework/hooks/useLogs.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        )}/${String(stepNum)}/stream`
+      )
+      eventSourceRef.current.onmessage = (event: MessageEvent<string>) => {
+        try {
+          if (event.type === LogsSSEEvent.MESSAGE) {
+            const newLog = JSON.parse(event.data)
+            setLogs((existingLogs: LivelogLine[]) => [...existingLogs, newLog])
+          }
+        } catch (_e) {
+          //ignore error
```
Comment (cjlee01) [inline apps/gitness/src/framework/hooks/useLogs.ts:50]: should we incorporate error with toast?

Comment (vardanbansal-harness) [reply apps/gitness/src/framework/hooks/useLogs.ts:50]: Will do that once we have toast working, I am yet to pick that up. It needs some more work.

File: apps/gitness/src/framework/hooks/useLogs.ts (hunk 1 above)
Comment (vardanbansal-harness) [reply apps/gitness/src/framework/hooks/useLogs.ts:22]: It's upto you and not a bad idea actually. `useSpaceSSE` currently handles more than one type of events so we can make a separate `usePREvents` that only handles events for pull requests.

File: apps/gitness/src/framework/hooks/useLogs.ts (hunk 2 above)
Comment (vardanbansal-harness) [reply apps/gitness/src/framework/hooks/useLogs.ts:13]: I am thinking we should only add reusable/common interfaces to the interface file, like we have `.types` file today in some directories. This interface is too specific to this hook. What do you think?

**Review by cjlee01 — APPROVED** [review]

**Pull request merged**